src/index.ts

```diff
@@ -1,207 +1,158 @@
-const RE_YOUTUBE =
-  /(?:youtube\.com\/(?:[^\/]+\/.+\/|(?:v|e(?:mbed)?)\/|.*[?&]v=)|youtu\.be\/)([^"&?\/\s]{11})/i;
+import { parse } from 'node-html-parser';
+
+const RE_PATH = /v|e(?:mbed)?|shorts/;
+
+const RE_CAPTION_TRACKS = /"captionTracks":\s?(\[.*?\])/;
+
+const ID_LENGTH = 11;
+
 const USER_AGENT =
   'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.83 Safari/537.36,gzip(gfe)';
 
-export class YoutubeTranscriptError extends Error {
-  constructor(message) {
-    super(`[YoutubeTranscript] 🚨 ${message}`);
-  }
+export interface TranscriptResponse {
+  text: string;
+  duration: number;
+  offset: number;
 }
 
 export interface TranscriptConfig {
+  /**
+   * Locale code
+   * @example en, es, hk, uk
+   */
   lang?: string;
-  country?: string;
 }
-export interface TranscriptResponse {
-  text: string;
-  duration: number;
-  offset: number;
+
+export class YoutubeTranscriptError extends Error {
```

Comment:
This can be improve using an response -:: [HTTP Problem Details (RFC 7807) ](https://www.npmjs.com/package/http-problem-details)